ui-tests/src/main/java/org/jenkinsci/test/acceptance/plugins/warnings_ng/IssuesTable.java

```diff
@@ -0,0 +1,168 @@
+package org.jenkinsci.test.acceptance.plugins.warnings_ng;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+
+import org.jenkinsci.test.acceptance.po.PageObject;
+
+/**
+ * Class representing an issues-table on the {@link AnalysisResult}.
+ *
+ * @author Stephan Plöderl
+ */
+public class IssuesTable {
+    private final AnalysisResult resultDetailsPage;
+    private final List<AbstractIssuesTableRow> tableRows = new ArrayList<>();
+    private final List<String> headers;
+    private final WebElement tableElement;
+    private final WebElement issuesTab;
+    private final IssuesTableRowType type;
+
+    /**
+     * Creates an IssuesTable of a specific type.
+     *
+     * @param issuesTab
+     *         the WebElement containing the issues-tab
+     * @param resultDetailsPage
+     *         the AnalysisResult on which the issues-table is displayed on
+     * @param type
+     *         the type of the issues-table (e.g. Default or DRY)
+     */
+    public IssuesTable(final WebElement issuesTab, final AnalysisResult resultDetailsPage, final IssuesTableRowType type) {
+        this.issuesTab = issuesTab;
+        this.resultDetailsPage = resultDetailsPage;
+        this.type = type;
+
+        tableElement = issuesTab.findElement(By.id("issues"));
+        headers = tableElement.findElements(By.xpath(".//thead/tr/th"))
+                .stream()
+                .map(WebElement::getText)
+                .collect(
+                        Collectors.toList());
+        updateTableRows();
+    }
+
+    /**
+     * Performs a click on a link on this site and returns the corresponding PageObject of the target page.
+     *
+     * @param link
+     *         the WebElement representing the link
+     * @param targetPageClass
+     *         the class of the PageObject representing the target page
+     *
+     * @return the PageObject representing the target page
+     */
+    public <T extends PageObject> T clickLinkOnSite(final WebElement link, final Class<T> targetPageClass) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Expected @param tag for '<T>'.](https://app.codacy.com/manual/uhafner/warnings-ng-plugin/pullRequest?prid=5070362)